utils/generate_js_fixtures.py

```diff
@@ -20,15 +24,15 @@ def build(srcdir: Path) -> None:
     subprocess.run(cmd, check=True, capture_output=True)
 
 
-for directory in TEST_JS_ROOTS.iterdir():
+for directory in TEST_JS_ROOTS:
     searchindex = directory / '_build' / 'searchindex.js'
     destination = TEST_JS_FIXTURES / directory.name / 'searchindex.js'
 
     print(f'Building {directory} ... ', end='')
     build(directory)
     print('done')
 
-    print(f'Moving {searchindex} to {destination} ... ', end='')
+    print(f'Copying {searchindex} to {destination} ... ', end='')
     destination.parent.mkdir(exist_ok=True)
-    searchindex.replace(destination)
+    destination.write_bytes(searchindex.read_bytes())
```

Comment:
Perhaps?
```suggestion
    shutil.copy2(searchindex, destination)
```

Reply:
Possibly?  I slightly prefer avoiding mix-and-match of library modules here, but a purpose-designed copy function is nice I admit.  I've no strong opinion either way.

Reply:
Purely for a utility script it doesn't really matter, but as ``shutil`` exposes a semantic copy operation, it is able to use [platform-dependent speedups](https://docs.python.org/3/library/shutil.html#shutil-platform-dependent-efficient-copy-operations) -- may be of interest!

Reply:
That is indeed of interest - as is fact that `copy2` attempts to retain file metadata - I wonder whether we should be using that instead of `shutil.copy` and `shutil.copyfile` in a few places in the codebase and tests.  A question for a separate thread, though..

The test search indexes should be small, and the file metadata doesn't matter in this case at at the moment, so I'm still feeling ambivalent (but thinking about it).

Reply:
Ok; there's not a lot in the decision, but it seems worth using the more efficient copy mechanism, given the choice between two functionally identical approaches.  If `copy` operations become available in `pathlib`, linting software might nudge us in that direction in future (or perhaps someone will remember / contribute a change to do so).

Reply:
(oops; I forgot that this also requires an `import`; fixing that up in a moment)